scanner/src/main/kotlin/Main.kt

```diff
@@ -193,8 +193,8 @@ object Main {
         // Make the parameter globally available.
         printStackTrace = stacktrace
 
-        require((dependenciesFile == null) != (inputPath == null)) {
-            "Either --dependencies-file or --input-path must be specified."
+        require((dependenciesFile == null) || (inputPath == null)) {
```

Comment:
This is not the same check anymore: Now having both `dependenciesFile = null` and `inputPath == null` passes the check, whereas previously it wouldn't (and shouldn't). If you want to use logical operators, this needs to be an [xor](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/xor.html). But I'm not sure whether that's really more readable than before: As we already use the `==` operator here, it seems more natural to also use `!=`. I'd actually prefer if this block stayed as-is.

Reply:
Ok, I just realized you're doing the missing check now in line 220. But I still feel the original code is more concise as it checks against the `dependenciesFile = null` and `inputPath == null` case early and in a single blow.

Reply:
The benefit is that the error messages with this change are more precise. Mainly I did this because I had to throw an exception at the end of this block:

```
val scanSummary: ScanSummary = dependenciesFile?.let {
    scanDependenciesFile(it)
} ?: inputPath?.let {
    scanInputPath(it)
} ?: throw IllegalArgumentException("Either --dependencies-file or --input-path must be specified.")
```

I could also do `inputPath!!...` because it must be not null at this point and keep the check as before.

Reply:
I just applied this change.